api-participant/src/main/java/bio/terra/pearl/api/participant/config/B2CConfigurationService.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.pearl.api.participant.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
```

Comment:
```suggestion
/** this is implemented as a service, rather than static config so that it may eventually support dynamic updates to the B2C configs */
```

Reply:
It mostly a service because we have to read a file, which is not standard config AFAIK. It also moves some logic out of the controller that should probably be an a service. Also, if B2C config gets more interesting in the future we will want a service (for instance, a better way to read local B2C config).
But I might be missing your point?

Reply:
yeah, Spring reading files for configuration is standard, and I think you're right that since this isn't just reading Spring-standard config files from spring-standard places it deserves a service

Reply:
Just curious, for "Spring reading files for configuration is standard" do you mean properties files or something else? One option I explored (and coded) for this was reading this as standard config via application.yml, but that config could have been pulled into that file from elsewhere. 
Is that what you mean, or something else?

Reply:
yup, that's what I meant--just reading properties from application.yaml would be better fit for a Configuration bean, but you're doing more here, because you have to support different ways of getting the file in